src/slic3r/GUI/MediaPlayCtrl.cpp

```diff
@@ -275,6 +276,12 @@ void MediaPlayCtrl::ToggleStream()
         bool need_install = false;
         if (!start_stream_service(&need_install)) {
             if (!need_install) return;
+#ifdef __LINUX__
+            (void) MessageDialog(this->GetParent(),
+                                 _L("This system does not have ffmpeg installed, which is required for Go Live to work.  Install it through your package manager, then try again."),
+                                 _L("Info"), wxOK).ShowModal();
+            return;
+#else
```

Comment:
@lanewei120 this comment is likely relevant to your internal solution, unless you're not using ffmpeg on *nix.

Reply:
I am not familiar with this
do you mean 'dpkg' or something else?

Reply:
Looks like this: https://www.freedesktop.org/software/PackageKit/files/session.c (https://www.freedesktop.org/software/PackageKit/pk-faq.html#session-methods)

New to me.  But I guess you'd want to use `InstallProvideFile("/usr/bin/ffmpeg")` or something?  It would be a very nice user experience, to be sure.

[Some people say that PackageKit is dying](https://blogs.gnome.org/hughsie/2019/02/14/packagekit-is-dead-long-live-well-something-else/).  I think those people have never actually tried to *use* a Snap-based distro on a day-to-day basis... but I digress.

Reply:
> I am not familiar with this do you mean 'dpkg' or something else?

PackageKit is an abstraction of `apk`, `dnf`, `apt/dpkg`, and others.
You can request specific software be installed at runtime, and PackageKit will make the appropriate package manager calls, and authentication needed to install.
> > The actual nuts-and-bolts distro tool (dnf, apt, etc) is used by PackageKit using compiled and scripted helpers. PackageKit isn't meant to replace these tools, instead providing a common set of abstractions that can be used by standard GUI and text mode package managers.

Sample usage can be found here: https://www.freedesktop.org/software/PackageKit/pk-using.html
under "Using libpackagekit"

The example makes sure `openoffice-clipart` is installed, and if it is not it will prompt the user to install it.

Reply:
I can try to add an example on top of this PR, but my C++ is very rusty, and basically limited to OS development :P

Reply:
The advantage here, is that if the user never tries to go live, this will never be installed. Reducing the footprint.

The downside is, that if the user does not have ffmpeg installed, they will need to install it at runtime. (But most linux distros do have it installed by default.)

Reply:
And it will also be hard to test, since support keeps asking me the same questions (that I've already answered).
Need replacement parts, been over a week and I'm not getting them :-1: just keep asking the same questions, and same troubleshooting steps.

Reply:
> Looks like this: https://www.freedesktop.org/software/PackageKit/files/session.c (https://www.freedesktop.org/software/PackageKit/pk-faq.html#session-methods)
> 
> New to me. But I guess you'd want to use `InstallProvideFile("/usr/bin/ffmpeg")` or something? It would be a very nice user experience, to be sure.
> 
> [Some people say that PackageKit is dying](https://blogs.gnome.org/hughsie/2019/02/14/packagekit-is-dead-long-live-well-something-else/). I think those people have never actually tried to _use_ a Snap-based distro on a day-to-day basis... but I digress.

A much nicer answer than I gave. Lol.

Reply:
Well, for the life of me, I can't convince PackageKit to do it.

On my machine, I did something like: `gdbus call -e -d org.freedesktop.PackageKit -o /org/freedesktop/PackageKit -m org.freedesktop.PackageKit.Modify.InstallPackageNames 1 '["lyx"]' "show-confirm-install,show-progress,show-finished" -i`

and absolutely nothing happened; `gdbus` just said `()`, and then that was that.

Reply:
1. Is it already installed?
2. Did you try provide file instead?